openff/toolkit/topology/topology.py

```diff
@@ -214,6 +214,107 @@ def __keytransform__(self, key):
         return key
 
 
+class UnsortedDict(_TransformedDict):
+    ...
+
+
+class TagSortedDict(_TransformedDict):
+    """
+    A dictionary where keys are kept unsorted, but only allows one permutation of a key
```

Comment:
```suggestion
    A dictionary where keys, consisting of tuples of atom indices, are kept unsorted, but only allows one permutation of a key
```

Something like this would be useful here, or just moved up to `_TransformedDict`